api/wasm/cpp/proxy_wasm_state.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Wasm State Class available to WASM modules.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <string>
+
+#include "envoy/stream_info/filter_state.h"
+
+// A simple wrapper around generic values
+class WasmState : public StreamInfo::FilterState::Object {
```

Comment:
I think we need some namespace for this?

Reply:
other classes in this directory are not under a namespace.. Ex: WasmData.. should we put this under namespace?

Reply:
WasmState should be usable in the rest of native envoy to be able to exchange data with native extensions. I think picking some Envoy-consistent namespace is necessary for it. Other stuff here I don't have a good reason to expose.

Reply:
Added Envoy::Wasm namespace... (am open to namespace name suggestions too : )

Reply:
The API should not have any namespaces.  We are trying to standardize this and it should not be Envoy specific.

Reply:
We add the namespaces when we include the file into Envoy (see wasm.cc and the way that proxy_wasm_result.h is included).